docs/src/apps/backwards-compatibility.md

```diff
@@ -0,0 +1,110 @@
+---
+title: Backwards Compatibility Policy
+---
+
+As the Solana developer ecosystem grows, so does the need for clear expectations around
+breaking API and behavior changes affecting applications and tooling built for Solana.
+In a perfect world, Solana development could continue at a very fast pace without ever
+causing issues for existing developers. However, some compromises will need to be made
+and so this document attempts to clarify and codify the process for new releases.
+
+### Expectations
+
+- Solana APIs, SDKs, and CLI tooling are tied to validator software releases (with a few
+  [exceptions](#exceptions)).
+- Software for a `MINOR` version release will be compatible across all software on the
+  **same** `MINOR` version _regardless_ of `PATCH` version.
```

Comment:
I think we should aim to stay compatible, at least, to the first `PATCH` release of the previous `MINOR` version.

Reply:
Sounds good, then any deprecated feature will be deprecated for a full MINOR release before being removed.

Reply:
I assume that once a feature has been deprecated in a MINOR version, the next MINOR version does not need to remain compatible with it, correct? If so, I think we should make that explicit. (If not, I don't follow how we ever remove things.)

Also, I think it is worth keeping the verbiage that says that the software will remain compatible across all PATCH releases in the same MINOR release.

Reply:
> I assume that once a feature has been deprecated in a MINOR version, the next MINOR version does not need to remain compatible with it, correct? If so, I think we should make that explicit. (If not, I don't follow how we ever remove things.)

Fully compatible back to at least two MINOR.0 versions in the past.   So the tip of master (1.5.0) should be fully compatible all the way back to 1.3.0+.   When we move to 1.6.0, at our option we can start removing features first deprecated in the 1.3.x release if/when they get in the way.

Reply:
Why at least two versions in the past? I think that one is sufficient so that things deprecated in 1.3.x can be removed in 1.5.0. At the current rate, a single MINOR release cycle is 2 months right? That means a change will be deprecated for 4-6 months if we support two versions rather than 2-4 months on one. The difference of 2 months just pushes back the inevitable for developers, they can expect breaking changes every MINOR.0 anyways.

Reply:
@mvines ping! I'm pushing back on the "at least 2 version" to just 1 version. Thoughts?

Reply:
I think the idea was to communicate that any deprecation will live on for _at least_ one full release cycle.  So we won't deprecate in the last patch of 1.4 and remove in 1.5.0

Reply:
I think we can expect at least some external devs to upgrade slowly, breaking them every 2-4 months is probably too frequent.  

> I think that one is sufficient so that things deprecated in 1.3.x can be removed in 1.5.0.
 
It's fair I think to mark things as deprecated the 1.3.x release while it remains on the stable channel.    This means what I marked as deprecated in 1.3.x last week and be removed from master today, and we'd then lose to ability to validate master against Serium.   From their perspective, we'd be asking them to remove a feature that's been deprecated for a week or less

Reply:
> I think we can expect at least some external devs to upgrade slowly, breaking them every 2-4 months is probably too frequent.

Hmm, how about breaking changes only on even numbered MINOR releases?

> This means what I marked as deprecated in 1.3.x last week could be removed from master today

Ah, putting it this way is interesting. Does that imply that we can't make breaking changes to beta?

Reply:
> Hmm, how about breaking changes only on even numbered MINOR releases?

Breaking changes are a major release bump?   Like what semver wants us to do anyway :)

> Does that imply that we can't make breaking changes to beta?

💯.

Reply:
> Breaking changes are a major release bump? Like what semver wants us to do anyway :)

Haha, that works!

Reply:
Alright, docs have been updated to promise proper semver 👍